cli/cli/helpers/grafloki/kubernetes_grafloki.go

```diff
@@ -0,0 +1,477 @@
+							Path:        "/ready",
+							Port:        intstr.FromInt(lokiPort),
+							Host:        "",
+							Scheme:      "",
+							HTTPHeaders: nil,
+						},
+						TCPSocket: nil,
+						GRPC:      nil,
+					},
+					InitialDelaySeconds:           lokiProbInitialDelaySeconds,
+					TimeoutSeconds:                lokiProbeTimeoutSeconds,
+					PeriodSeconds:                 lokiProbePeriodSeconds,
+					SuccessThreshold:              0,
+					FailureThreshold:              0,
+					TerminationGracePeriodSeconds: nil,
+				},
+				StartupProbe:             nil,
+				Lifecycle:                nil,
+				TerminationMessagePath:   "",
+				TerminationMessagePolicy: "",
+				ImagePullPolicy:          "",
+				SecurityContext:          nil,
+				Stdin:                    false,
+				StdinOnce:                false,
+				TTY:                      false,
+			},
+		},
+		[]apiv1.Volume{},
+		&apiv1.Affinity{
+			NodeAffinity:    nil,
+			PodAffinity:     nil,
+			PodAntiAffinity: nil,
+		})
+	if err != nil {
+		return "", stacktrace.Propagate(err, "An error occurred creating Loki deployment.")
+	}
+	shouldRemoveLokiDeployment := false
+	defer func() {
+		if shouldRemoveLokiDeployment {
+			if err := k8sManager.RemoveDeployment(ctx, graflokiNamespace, lokiDeployment); err != nil {
+				logrus.Warnf("Attempted to remove Loki deployment after an error occurred creating it but an error occurred removing it.")
+				logrus.Warnf("Manually remove Loki deployment with name: %v", lokiDeployment.Name)
+			}
+		}
+	}()
+	logrus.Infof("Waiting for Loki deployment to come online (can take around 30s)... ")
+	if err := k8sManager.WaitForPodManagedByDeployment(ctx, lokiDeployment, lokiDeploymentMaxRetires, lokiDeploymentRetryInterval); err != nil {
+		return "", stacktrace.Propagate(err, "An error occurred while waiting for pod managed by Loki deployment '%v' to come online.", lokiDeploymentName)
+	}
+
+	lokiService, err := k8sManager.CreateService(ctx,
+		graflokiNamespace,
+		lokiServiceName,
+		map[string]string{}, // empty labels
+		map[string]string{}, // empty annotations
+		lokiLabels,          // match loki deployment pod labels
+		apiv1.ServiceTypeNodePort,
+		[]apiv1.ServicePort{{
+			Name:        "logs-listening",
+			Port:        lokiNodePort,
+			TargetPort:  intstr.FromInt(lokiPort),
+			Protocol:    apiv1.ProtocolTCP,
+			NodePort:    lokiNodePort,
+			AppProtocol: &httpApplicationProtocol,
+		}})
+	shouldRemoveLokiService := false
+	defer func() {
+		if shouldRemoveLokiService {
+			if err := k8sManager.RemoveService(ctx, lokiService); err != nil {
+				logrus.Warnf("Attempted to remove Loki service after an error occurred creating it but an error occurred removing it.")
+				logrus.Warnf("Manually remove Loki service with name: %v", lokiService.Name)
+			}
+		}
+	}()
+	if err != nil {
+		return "", stacktrace.Propagate(err, "An error occurred creating Loki service")
+	}
+	lokiHost := getLokiHost(lokiServiceName, graflokiNamespace, lokiNodePort)
```

Comment:
So generally Kubernetes doesn't want you getting actual host (node) names. You _can_, but the more canonical way is to get the domain name of the service (which lets k8s do any migrations and such that it deems necessary transparently to the consumer)